parkings/api/operator/permit.py

```diff
@@ -0,0 +1,29 @@
+from ..common_permit import PermitSerializer, PermitViewSet
+from .operator_permission import OperatorApiHasPermission
+
+
+class DomainNotRequiredMixin():
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields['domain'].required = False
+
+
+class DomainToCodeMixin():
+    def to_representation(self, instance):
+        representation = super().to_representation(instance)
+        representation.update(domain=instance.domain.code)
```

Comment:
Use `SlugRelatedField` for the `domain` field so you don't have to do this conversion manually and it will handle the other way also. https://www.django-rest-framework.org/api-guide/relations/#slugrelatedfield

Reply:
```python
class OperatorPermitSerializer(...):
    domain = serializers.SlugRelatedField(
        slug_field='code', queryset=EnforcementDomain.objects.all())
```